ls1intum/Artemis
# [Feature] Overview for students' questions

### Checklist

### Motivation and Context


Tutors and instructors are not able to see all questions for a course in one place. In order to see if there are unanswered questions they have to check each exercise and lecture individually.

### Description

Added an overview for tutors and instructors to see all questions in a course. By clicking the question's ID they can navigate to the question and see the answers.

### Steps for Testing


**Setup:** Make sure that you have a test course with some exercises and lectures. Each should have some questions and some answers. Try to check as many combinations as possible.

1. Log in to Artemis
2. Navigate to Course Management
3. Click on the _Q&A_ button
4. Check that all questions are displayed
5. Check that the amounts of answers and approved answers is correct
6. Check that you can navigate to the correct exercise /lecture by clicking the question ID or the title of the exercise / lecture.

### Test Coverage

* StudentQuestionRessource.java: 84%
* course-questions.component.ts 85%

### Screenshots



Button in course view:
<img width="309" alt="Bildschirmfoto 2020-10-08 um 20 57 31" src="https://user-images.githubusercontent.com/51987021/95502230-ca1db680-09a9-11eb-916b-8222a34ba45d.png">

Button in tutor dashboard:
<img width="673" alt="Bildschirmfoto 2020-10-08 um 20 59 24" src="https://user-images.githubusercontent.com/51987021/95502281-e588c180-09a9-11eb-958f-98f5c667caf9.png">

Link in instructor dashboard:
<img width="319" alt="Bildschirmfoto 2020-10-08 um 20 59 33" src="https://user-images.githubusercontent.com/51987021/95502305-ee799300-09a9-11eb-8f5d-a7efb8a3281e.png">



Question overview:
<img width="1430" alt="Bildschirmfoto 2020-09-29 um 21 14 46" src="https://user-images.githubusercontent.com/51987021/94605062-d8c4e900-0298-11eb-904d-16e68c3ead9c.png">

<img width="1410" alt="Bildschirmfoto 2020-09-29 um 21 18 17" src="https://user-images.githubusercontent.com/51987021/94605376-525cd700-0299-11eb-979d-39564017e480.png">
